### ISSUE TITLE
Handle pre-upload BulkUploadFilePicker errors

### ISSUE THREAD

**Issue by reefdog** [PhilanthropyDataCommons/front-end#528]
#519 adds basic bulk uploads, but it's lacking error handling when the upload fails during initial registration with the API (or any other error prior to the API receiving the upload).

In this case, we should set the `status` state of BulkUploadFilePicker to `error.{someErrorType}` and add the appropriate case and content to [this render block](https://github.com/PhilanthropyDataCommons/front-end/blob/d5529b4bcbc4c36dfaf3719c794030995e1827bf/src/components/BulkUploaderFilePicker.tsx#L275-L315).

**Comment by reefdog** [comment]
This will actually be handled by #519 after all (thanks @slifty!)

https://github.com/PhilanthropyDataCommons/front-end/pull/519/files#diff-41f7ffeecdd48d4e454dcd23aca562e2ca5206882673a15f18467d97dfeba969R134-R136

Since we don't really know *why* the upload failed, we fall through to the default error response. But I've removed the `error.generic` case since we weren't using this, and it's handled by `default` anyway. I did go ahead and add `error.uploadFailure` to the list before `default` — even though it falls through — because I like the explicitness of this.

https://github.com/PhilanthropyDataCommons/front-end/pull/519/files#diff-41f7ffeecdd48d4e454dcd23aca562e2ca5206882673a15f18467d97dfeba969L305-R319
